c2_components/src/mfx_c2_decoder_component.cpp

```diff
@@ -1666,6 +1667,94 @@ c2_status_t MfxC2DecoderComponent::AllocateFrame(MfxC2FrameOut* frame_out)
     return res;
 }
 
+bool MfxC2DecoderComponent::IsDuplicatedTimeStamp(uint64_t timestamp)
+{
+    MFX_DEBUG_TRACE_FUNC;
+
+    bool bDuplicated = false;
+
+    auto duplicate = find_if(m_duplicatedTimeStamp.begin(), m_duplicatedTimeStamp.end(),
+        [timestamp] (const auto &item) {
+            return item.first == timestamp;
+    });
+
+    if (duplicate != m_duplicatedTimeStamp.end()) {
+        bDuplicated = true;
+        MFX_DEBUG_TRACE_STREAM("Potentional error: Found duplicated timestamp: "
+                        << duplicate->first);
+    }
+
+    return bDuplicated;
+}
+
+bool MfxC2DecoderComponent::IsPartialFrame(uint64_t frame_index)
+{
+    MFX_DEBUG_TRACE_FUNC;
+
+    bool bDuplicated = false;
+
+    auto duplicate = find_if(m_duplicatedTimeStamp.begin(), m_duplicatedTimeStamp.end(),
+        [frame_index] (const auto &item) {
+            return item.second == frame_index;
+    });
+
+    if (duplicate != m_duplicatedTimeStamp.end()) {
+        bDuplicated = true;
+        MFX_DEBUG_TRACE_STREAM("Potentional error: Found duplicated timestamp: "
+                        << duplicate->first);
+    }
+
+    return bDuplicated;
+}
+
+void MfxC2DecoderComponent::EmptyReadViews(uint64_t timestamp, uint64_t frame_index)
+{
+    MFX_DEBUG_TRACE_FUNC;
+    MFX_DEBUG_TRACE_I64(timestamp);
+    MFX_DEBUG_TRACE_I64(frame_index);
+
+    if (0)
```

Comment:
can we remove this condition block if not needed.

Reply:
I keep it for debugging purpose. But it looks ugly. We can remove it.